feature/s3/manager/bucket_region.go

```diff
@@ -63,20 +63,21 @@ const bucketRegionHeader = "X-Amz-Bucket-Region"
 //
 //	manager.GetBucketRegion(ctx, s3.NewFromConfig(cfg), bucket, func(o *s3.Options) {
 //	     o.Credentials = nil
-//	     // Or
-//	     o.Credentials = aws.AnonymousCredentials{}
 //	})
 //
 // The request with anonymous credentials will not be signed.
 // Otherwise credentials would be required for private buckets.
 func GetBucketRegion(ctx context.Context, client HeadBucketAPIClient, bucket string, optFns ...func(*s3.Options)) (string, error) {
 	var captureBucketRegion deserializeBucketRegion
 
-	clientOptionFns := make([]func(*s3.Options), len(optFns)+1)
+	clientOptionFns := make([]func(*s3.Options), len(optFns)+2)
 	clientOptionFns[0] = func(options *s3.Options) {
 		options.APIOptions = append(options.APIOptions, captureBucketRegion.RegisterMiddleware)
 	}
-	copy(clientOptionFns[1:], optFns)
+	clientOptionFns[1] = func(options *s3.Options) {
+		options.Credentials = nil
+	}
+	copy(clientOptionFns[2:], optFns)
```

Comment:
Thanks for fixing the bug!

Out of curiosity, is there a reason why we don't just set `options.Credentials = nil `directly in `clientOptionFns[0]`?

```suggestion
	clientOptionFns[0] = func(options *s3.Options) {
		options.APIOptions = append(options.APIOptions, captureBucketRegion.RegisterMiddleware)
		options.Credentials = nil
	}
	copy(clientOptionFns[1:], optFns)
```

Reply:
It doesn't matter whether you put it at index 0 or 1 as long as it is executed ahead other functional config (e.g. user want to reset their credentials using a option fn)